utils/zerovec/src/map/map.rs

```diff
@@ -321,6 +321,84 @@ where
     }
 }
 
+impl<'a, K, V> ZeroMap<'a, K, V>
+where
+    K: ZeroMapKV<'a, Container = ZeroVec<'a, K>> + ?Sized,
+    V: ZeroMapKV<'a> + ?Sized,
+    K: AsULE,
+{
+    /// Cast a `ZeroMap<K, V>` to `ZeroMap<P, V>` where `K` and `P` are [`AsULE`] types
+    /// with the same representation.
+    ///
+    /// If `K` and `P` have different ordering semantics, unexpected behavior may occur.
+    pub fn cast_zv_k<P>(self) -> ZeroMap<'a, P, V>
```

Comment:
Thought: Maybe these should be prefixed with `unchecked` since there is a (non-safety) invariant that must be upheld

Reply:
suffix with unchecked is probably better

Reply:
but yeah good idea